src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -82,7 +89,19 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
 
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor, shouldMerge);
+        if (shouldMerge) {
+            // If merge and tag prefix is present but input is empty
+            if (!editPersonDescriptor.getTags().isEmpty() && editPersonDescriptor.getTags().get().isEmpty()) {
+                throw new CommandException(MESSAGE_NO_TAG_ADDED);
+            }
+
+            // If merge and group prefix is present but input is empty
+            if (!editPersonDescriptor.getGroups().isEmpty() && editPersonDescriptor.getGroups().get().isEmpty()) {
+                throw new CommandException(MESSAGE_NO_GROUP_ADDED);
+            }
+        }
```

Comment:
I'm thinking if the checking of the fields should be done in the EditCommandParser instead?

Reply:
Kind of nitpicking...